File: frontend/js/reportes.js
async function cargarReportes() {
    const seccion = document.getElementById('reportes');
    seccion.innerHTML = `
        <div class="page-header">
            <h2>Reportes</h2>
            <p>Resumen general del sistema de actividades deportivas</p>
        </div>
        <div id="reportes-contenido">
            <div style="text-align:center;padding:40px;color:#9ca3af">
                <i class="ti ti-loader" style="font-size:32px;display:block;margin-bottom:8px"></i>
                Cargando reportes...
            </div>
        </div>
    `;

    const [
        resInscriptos, resCupos, resDisciplina, resCarrera,
        resOcupacion, resAsistencia, resInasistencias,
        resEspera, resActivos, resVacias
    ] = await Promise.all([
        fetch(`${API}/reportes/inscriptos-por-actividad`),
        fetch(`${API}/reportes/cupos-disponibles`),
        fetch(`${API}/reportes/inscriptos-por-disciplina`),
        fetch(`${API}/reportes/inscriptos-por-carrera`),
        fetch(`${API}/reportes/ocupacion`),
        fetch(`${API}/reportes/asistencia`),
        fetch(`${API}/reportes/inasistencias`),
        fetch(`${API}/reportes/lista-espera`),
        fetch(`${API}/reportes/estudiantes-activos`),
        fetch(`${API}/reportes/actividades-vacias`)
    ]);

    const inscriptos    = await resInscriptos.json();
    const cupos         = await resCupos.json();
    const disciplina    = await resDisciplina.json();
    const carrera       = await resCarrera.json();
    const ocupacion     = await resOcupacion.json();
    const asistencia    = await resAsistencia.json();
    const inasistencias = await resInasistencias.json();
    const espera        = await resEspera.json();
    const activos       = await resActivos.json();
    const vacias        = await resVacias.json();

    const totalConfirmados = inscriptos.reduce((s, r) => s + r.total_confirmados, 0);
    const avgAsistencia    = asistencia.length
        ? Math.round(asistencia.reduce((s, r) => s + (parseFloat(r.porcentaje_asistencia) || 0), 0) / asistencia.length)
        : 0;

    document.getElementById('reportes-contenido').innerHTML = `

        <div class="stats-grid" style="grid-template-columns:repeat(4,1fr);margin-bottom:28px">
            <div class="stat-card">
                <div class="stat-icon"><i class="ti ti-clipboard-check"></i></div>
                <div class="stat-num">${totalConfirmados}</div>
                <div class="stat-lbl">Inscripciones confirmadas</div>
            </div>
            <div class="stat-card">
                <div class="stat-icon"><i class="ti ti-clock" style="color:#92400e"></i></div>
                <div class="stat-num" style="color:#92400e">${espera.length}</div>
                <div class="stat-lbl">En lista de espera</div>
            </div>
            <div class="stat-card">
                <div class="stat-icon"><i class="ti ti-chart-bar" style="color:#166534"></i></div>
                <div class="stat-num" style="color:#166534">${avgAsistencia}%</div>
                <div class="stat-lbl">Asistencia promedio</div>
            </div>
            <div class="stat-card">
                <div class="stat-icon"><i class="ti ti-alert-triangle" style="color:#991b1b"></i></div>
                <div class="stat-num" style="color:#991b1b">${inasistencias.length}</div>
                <div class="stat-lbl">Estudiantes con 3+ faltas</div>
            </div>
        </div>

        <div style="display:grid;grid-template-columns:1fr 1fr;gap:18px;margin-bottom:18px">
            ${tablaReporte('Inscriptos por actividad', ['Actividad', 'Confirmados'],
                inscriptos.map(r => `
                    <tr>
                        <td>${r.actividad}</td>
                        <td>
                            <div style="display:flex;align-items:center;gap:8px">
                                <div style="background:#e8ecf0;border-radius:20px;height:5px;width:60px;flex-shrink:0">
                                    <div style="background:#1a56c4;height:5px;border-radius:20px;width:${Math.min(r.total_confirmados * 20, 100)}%"></div>
                                </div>
                                <span style="font-weight:600;color:#0a2e6e">${r.total_confirmados}</span>
                            </div>
                        </td>
                    </tr>`)
            )}
            ${tablaReporte('Inscriptos por disciplina', ['Disciplina', 'Inscriptos'],
                disciplina.map(r => `
                    <tr>
                        <td>${r.disciplina}</td>
                        <td style="font-weight:600;color:#0a2e6e">${r.total_inscriptos}</td>
                    </tr>`)
            )}
        </div>

        <div style="display:grid;grid-template-columns:1fr 1fr;gap:18px;margin-bottom:18px">
            ${tablaReporte('Ocupación por actividad', ['Actividad', 'Cupo', 'Confirmados', 'Ocupación'],
                ocupacion.map(r => {
                    const pct = parseFloat(r.porcentaje_ocupacion) || 0;
                    const color = pct >= 100 ? '#ef4444' : pct >= 70 ? '#f59e0b' : '#1a56c4';
                    return `
                    <tr>
                        <td>${r.actividad}</td>
                        <td style="color:#6b7280">${r.cupo_maximo}</td>
                        <td style="color:#6b7280">${r.confirmados}</td>
                        <td>
                            <div style="display:flex;align-items:center;gap:6px">
                                <div style="background:#e8ecf0;border-radius:20px;height:5px;width:50px;flex-shrink:0">
                                    <div style="background:${color};height:5px;border-radius:20px;width:${Math.min(pct,100)}%"></div>
                                </div>
                                <span style="font-size:11px;font-weight:600;color:${color}">${pct}%</span>
                            </div>
                        </td>
                    </tr>`
                })
            )}
            ${tablaReporte('Asistencia por actividad', ['Actividad', 'Clases', 'Presentes', 'Asistencia'],
                asistencia.map(r => {
                    const pct = parseFloat(r.porcentaje_asistencia) || 0;
                    const color = pct >= 80 ? '#166534' : pct >= 50 ? '#92400e' : '#991b1b';
                    return `
                    <tr>
                        <td>${r.actividad}</td>
                        <td style="color:#6b7280">${r.clases_registradas}</td>
                        <td style="color:#6b7280">${r.presentes}</td>
                        <td style="font-weight:600;color:${color}">${pct}%</td>
                    </tr>`
                })
            )}
        </div>

        <div style="display:grid;grid-template-columns:1fr 1fr;gap:18px;margin-bottom:18px">
            ${tablaReporte('Lista de espera', ['Actividad', 'Estudiante', 'Fecha'],
                espera.length === 0
                    ? [`<tr><td colspan="3" style="text-align:center;color:#9ca3af;padding:20px">Sin estudiantes en espera</td></tr>`]
                    : espera.map(r => `
                        <tr>
                            <td>${r.actividad}</td>
                            <td style="font-weight:500">${r.nombre} ${r.apellido}</td>
                            <td style="color:#6b7280">${new Date(r.fecha_en_espera).toLocaleDateString('es-UY')}</td>
                        </tr>`)
            )}
            ${tablaReporte('Estudiantes con 3+ inasistencias', ['Estudiante', 'Actividad', 'Faltas'],
                inasistencias.length === 0
                    ? [`<tr><td colspan="3" style="text-align:center;color:#9ca3af;padding:20px">Sin estudiantes con 3+ inasistencias</td></tr>`]
                    : inasistencias.map(r => `
                        <tr>
                            <td style="font-weight:500">${r.nombre} ${r.apellido}</td>
                            <td style="color:#6b7280">${r.actividad}</td>
                            <td><span class="badge badge-cancelada">${r.total_inasistencias}</span></td>
                        </tr>`)
            )}
        </div>

        <div style="display:grid;grid-template-columns:1fr 1fr;gap:18px;margin-bottom:18px">
            ${tablaReporte('Estudiantes más activos', ['Estudiante', 'Carrera', 'Actividades'],
                activos.map(r => `
                    <tr>
                        <td style="font-weight:500">${r.nombre} ${r.apellido}</td>
                        <td style="color:#6b7280">${r.carrera}</td>
                        <td><span class="badge badge-confirmada">${r.actividades_confirmadas}</span></td>
                    </tr>`)
            )}
            ${tablaReporte('Actividades sin inscriptos', ['Actividad', 'Disciplina', 'Estado'],
                vacias.length === 0
                    ? [`<tr><td colspan="3" style="text-align:center;color:#9ca3af;padding:20px">Todas las actividades tienen inscriptos</td></tr>`]
                    : vacias.map(r => `
                        <tr>
                            <td style="font-weight:500">${r.actividad}</td>
                            <td style="color:#6b7280">${r.disciplina}</td>
                            <td><span class="badge badge-${r.estado}">${r.estado}</span></td>
                        </tr>`)
            )}
        </div>

        ${tablaReporte('Inscriptos por carrera y facultad', ['Facultad', 'Carrera', 'Inscriptos'],
            carrera.map(r => `
                <tr>
                    <td style="color:#6b7280">${r.facultad}</td>
                    <td style="font-weight:500">${r.carrera}</td>
                    <td style="font-weight:600;color:#0a2e6e">${r.total_inscriptos}</td>
                </tr>`),
            true
        )}
    `;
}

function tablaReporte(titulo, columnas, filas, fullWidth = false) {
    return `
        <div class="reporte-card" ${fullWidth ? 'style="grid-column:1/-1"' : ''}>
            <h3>${titulo}</h3>
            <div style="overflow-x:auto">
                <table>
                    <thead>
                        <tr>${columnas.map(c => `<th>${c}</th>`).join('')}</tr>
                    </thead>
                    <tbody>
                        ${filas.join('')}
                    </tbody>
                </table>
            </div>
        </div>
    `;
}

File: backend/routes/reportes.py
from flask import Blueprint, jsonify
from database import get_connection

reportes_bp = Blueprint('reportes', __name__)

# 1. Actividades con mayor cantidad de inscriptos confirmados
@reportes_bp.route('/reportes/inscriptos-por-actividad', methods=['GET'])
def inscriptos_por_actividad():
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("""
        SELECT a.nombre AS actividad,
               COUNT(i.id_inscripcion) AS total_confirmados
        FROM actividad a
        LEFT JOIN inscripcion i ON a.id_actividad = i.id_actividad
                               AND i.estado = 'confirmada'
        GROUP BY a.id_actividad, a.nombre
        ORDER BY total_confirmados DESC
    """)
    resultado = cursor.fetchall()
    cursor.close()
    conn.close()
    return jsonify(resultado)

# 2. Actividades con cupos disponibles
@reportes_bp.route('/reportes/cupos-disponibles', methods=['GET'])
def cupos_disponibles():
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("""
        SELECT a.nombre AS actividad, a.cupo_maximo,
               COUNT(i.id_inscripcion) AS confirmados,
               a.cupo_maximo - COUNT(i.id_inscripcion) AS cupos_disponibles
        FROM actividad a
        LEFT JOIN inscripcion i ON a.id_actividad = i.id_actividad
                               AND i.estado = 'confirmada'
        WHERE a.estado = 'abierta'
        GROUP BY a.id_actividad, a.nombre, a.cupo_maximo
        HAVING cupos_disponibles > 0
        ORDER BY cupos_disponibles DESC
    """)
    resultado = cursor.fetchall()
    cursor.close()
    conn.close()
    return jsonify(resultado)

# 3. Cantidad de inscriptos por disciplina
@reportes_bp.route('/reportes/inscriptos-por-disciplina', methods=['GET'])
def inscriptos_por_disciplina():
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("""
        SELECT d.nombre AS disciplina,
               COUNT(i.id_inscripcion) AS total_inscriptos
        FROM disciplina d
        LEFT JOIN actividad a   ON d.id_disciplina = a.id_disciplina
        LEFT JOIN inscripcion i ON a.id_actividad  = i.id_actividad
                               AND i.estado = 'confirmada'
        GROUP BY d.id_disciplina, d.nombre
        ORDER BY total_inscriptos DESC
    """)
    resultado = cursor.fetchall()
    cursor.close()
    conn.close()
    return jsonify(resultado)

# 4. Cantidad de inscriptos por carrera y facultad
@reportes_bp.route('/reportes/inscriptos-por-carrera', methods=['GET'])
def inscriptos_por_carrera():
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("""
        SELECT e.facultad, e.carrera,
               COUNT(i.id_inscripcion) AS total_inscriptos
        FROM estudiante e
        LEFT JOIN inscripcion i ON e.id_estudiante = i.id_estudiante
                               AND i.estado = 'confirmada'
        GROUP BY e.facultad, e.carrera
        ORDER BY e.facultad, total_inscriptos DESC
    """)
    resultado = cursor.fetchall()
    cursor.close()
    conn.close()
    return jsonify(resultado)

# 5. Porcentaje de ocupación por actividad
@reportes_bp.route('/reportes/ocupacion', methods=['GET'])
def ocupacion():
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("""
        SELECT a.id_actividad, a.nombre AS actividad, a.cupo_maximo,
               COUNT(i.id_inscripcion) AS confirmados,
               ROUND(COUNT(i.id_inscripcion) * 100.0 / a.cupo_maximo, 1) AS porcentaje_ocupacion
        FROM actividad a
        LEFT JOIN inscripcion i ON a.id_actividad = i.id_actividad
                               AND i.estado = 'confirmada'
        GROUP BY a.id_actividad, a.nombre, a.cupo_maximo
        ORDER BY porcentaje_ocupacion DESC
    """)
    resultado = cursor.fetchall()
    cursor.close()
    conn.close()
    return jsonify(resultado)

# 6. Porcentaje de asistencia por actividad
@reportes_bp.route('/reportes/asistencia', methods=['GET'])
def asistencia():
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("""
        SELECT a.nombre AS actividad,
               COUNT(ast.id_asistencia) AS clases_registradas,
               SUM(ast.presente) AS presentes,
               ROUND(SUM(ast.presente) * 100.0 / NULLIF(COUNT(ast.id_asistencia), 0), 1) AS porcentaje_asistencia
        FROM actividad a
        JOIN inscripcion i  ON a.id_actividad   = i.id_actividad
                           AND i.estado         = 'confirmada'
        JOIN asistencia ast ON i.id_inscripcion = ast.id_inscripcion
        GROUP BY a.id_actividad, a.nombre
        ORDER BY porcentaje_asistencia DESC
    """)
    resultado = cursor.fetchall()
    cursor.close()
    conn.close()
    return jsonify(resultado)

# 7. Estudiantes con 3 o más inasistencias
@reportes_bp.route('/reportes/inasistencias', methods=['GET'])
def inasistencias():
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("""
        SELECT e.nombre, e.apellido, a.nombre AS actividad,
               COUNT(ast.id_asistencia) AS total_inasistencias
        FROM estudiante e
        JOIN inscripcion i  ON e.id_estudiante  = i.id_estudiante
        JOIN actividad a    ON i.id_actividad   = a.id_actividad
        JOIN asistencia ast ON i.id_inscripcion = ast.id_inscripcion
        WHERE ast.presente = FALSE
        GROUP BY e.id_estudiante, e.nombre, e.apellido, a.id_actividad, a.nombre
        HAVING total_inasistencias >= 3
        ORDER BY total_inasistencias DESC
    """)
    resultado = cursor.fetchall()
    cursor.close()
    conn.close()
    return jsonify(resultado)

# 8. Estudiantes en lista de espera por actividad
@reportes_bp.route('/reportes/lista-espera', methods=['GET'])
def lista_espera():
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("""
        SELECT a.nombre AS actividad, e.nombre, e.apellido,
               i.fecha_inscripcion AS fecha_en_espera
        FROM inscripcion i
        JOIN estudiante e ON i.id_estudiante = e.id_estudiante
        JOIN actividad  a ON i.id_actividad  = a.id_actividad
        WHERE i.estado = 'en_espera'
        ORDER BY a.nombre, i.fecha_inscripcion ASC
    """)
    resultado = cursor.fetchall()
    cursor.close()
    conn.close()
    return jsonify(resultado)

# 9. Ranking de estudiantes más activos
@reportes_bp.route('/reportes/estudiantes-activos', methods=['GET'])
def estudiantes_activos():
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("""
        SELECT e.nombre, e.apellido, e.carrera,
               COUNT(i.id_inscripcion) AS actividades_confirmadas
        FROM estudiante e
        JOIN inscripcion i ON e.id_estudiante = i.id_estudiante
                          AND i.estado = 'confirmada'
        GROUP BY e.id_estudiante, e.nombre, e.apellido, e.carrera
        ORDER BY actividades_confirmadas DESC
    """)
    resultado = cursor.fetchall()
    cursor.close()
    conn.close()
    return jsonify(resultado)

# 10. Actividades sin inscriptos confirmados
@reportes_bp.route('/reportes/actividades-vacias', methods=['GET'])
def actividades_vacias():
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)
    cursor.execute("""
        SELECT a.nombre AS actividad, d.nombre AS disciplina,
               a.estado, a.dia, a.horario
        FROM actividad a
        JOIN disciplina d ON a.id_disciplina = d.id_disciplina
        WHERE a.id_actividad NOT IN (
            SELECT DISTINCT id_actividad FROM inscripcion
            WHERE estado = 'confirmada'
        )
        ORDER BY a.nombre
    """)
    resultado = cursor.fetchall()
    cursor.close()
    conn.close()
    return jsonify(resultado)